Assets/Scripts/PlayerHat.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using Photon.Pun;
+
+public class PlayerHat : MonoBehaviourPunCallbacks {
+    private Vector3 parentPos;
+
+    void Start() {
+        if (PhotonNetwork.IsConnected && !photonView.IsMine) {
```

Comment:
the hats don't have a photonView attached to them and this line will always throw null exception. you can either remove the whole `Start()` (since it is ok for the script to be enabled as the parent's transform is synced via photonview) or us `GetComponentInParent<PhotonView>()`